pycolleff/pycolleff/colleff.py

```diff
@@ -379,22 +421,51 @@ def kick_factor(
         energy = self.energy
         w0 = self.rev_ang_freq
         nb = self.num_bun
+        hnum = self.harm_num
         bunlen = self.bunlen
         tot_curr = self.total_current
 
         w, Z = self._prepare_input_impedance(budget, element, w, Z, imp_type)
 
         limw = 5 * _LSPEED / bunlen
-        w[-1] = min(w[-1], limw)
-        w[0] = max(w[0], -limw)
-        wp = self._get_sampling_ang_freq(w, w0, nb, 0, 0, [0], nut=nut)
+        wmin = max(w[0], -limw)
+        wmax = min(w[-1], limw)
+
+        if fillpattern is not None:
+            if fillpattern.ndim != 1:
```

Comment:
Maybe you could create a function to verify if the filllpattern is valid? I don't know if there is a gain in doing it, I am just suggesting it because this code is repeated two times.

Reply:
yes, I thought about that, and didn't implement it because I was lazy :laughing: 
I'll implement the suggestion! Thanks!

Reply:
done, @Gabrielrezende-asc , thanks!